sdk/Sdk.Analyzers/AsyncVoidAnalyzer.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class AsyncVoidAnalyzer : DiagnosticAnalyzer
+    {    
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DiagnosticDescriptors.AsyncVoidReturnType);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+
+            context.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Method);
+        }
+
+        private static void AnalyzeMethod(SymbolAnalysisContext symbolAnalysisContext)
+        {
+            var symbol = (IMethodSymbol)symbolAnalysisContext.Symbol;
+
+            if (symbol.IsAsync && symbol.ReturnsVoid)
+            {
+                var diagnostic = Diagnostic.Create(DiagnosticDescriptors.AsyncVoidReturnType, symbol.Locations[0]);
```

Comment:
Definitely a nit pick here, but I think it might be nicer to avoid a magic number such as `symbol.Locations[0]` in code; is there a reason this will always be `0`? would it make sense to make this a variable or add a comment around it?

Reply:
Good point. Based on the [docs](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.isymbol.locations?view=roslyn-dotnet-3.11.0), the `ISymbol.Locations` property is an array and it will have more than one entry in cases like partial classes where you will have more than one place where the symbol is located. 

The Location property will have at-least one item in the array all the time (where the symbol we are analyzing is located). In our case, we already narrowed it down to a method and it is guaranteed to be of size 1. I could not find a use case where it will have more than one locations (I am curious about the local function use case. I will investigate that). I will do a bit more research to see there is a more readable API/Extension method available to use. I left a comment in the file for time being, as you suggested.